### PR TITLE
geant4: add hdf5 variant

### PR DESCRIPTION
**Pull request by DingXuefeng**
Hi, I added variant +hdf to allow geant4 to be compiled with support for HDF5 output format.

@drbenmorgan @sethrj

### CODE REVIEW

**Review by drbenmorgan — DISMISSED** [review]
Looks good to me, thanks @DingXuefeng!

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 1]
```diff
@@ -78,6 +78,7 @@ class Geant4(CMakePackage):
     variant("x11", default=False, description="Optional X11 support")
     variant("motif", default=False, description="Optional motif support")
     variant("qt", default=False, description="Enable Qt support")
+    variant("hdf", default=False, description="Enable HDF5 support", when="@10.4:")
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/geant4/package.py]: Can you spell this as `hdf5` to be more consistent with the other (dozens of) Spack packages?

```suggestion
    variant("hdf5", default=False, description="Enable HDF5 support", when="@10.4:")
```

File: var/spack/repos/builtin/packages/geant4/package.py [hunk 2]
```diff
@@ -316,6 +320,9 @@ def cmake_args(self):
                 options.append(self.define("GEANT4_USE_QT_QT6", True))
             options.append(self.define("QT_QMAKE_EXECUTABLE", spec["qmake"].prefix.bin.qmake))
 
+        if "+hdf" in spec:
+            options.append(self.define("GEANT4_USE_HDF5", True))
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/geant4/package.py]: This should work and be more foolproof (explicitly setting it to "off" if disabled)
```suggestion
        options.append(self.define_from_variant("GEANT4_USE_HDF5", "hdf"))
```

**Review by sethrj — CHANGES_REQUESTED** [review]

**Review by sethrj — APPROVED** [review]
Thanks!

**Pull request merged**